src/WorkspaceDependencyViewExtension/WorkspaceDependencyView.xaml.cs

```diff
@@ -170,137 +118,112 @@ internal void DependencyRegen(WorkspaceModel ws)
-        /// Constructor
-        /// </summary>
-        /// <param name="data"></param>
-        /// <param name="color"></param>
-        public ColumnData(string data, Brush color)
-        {
-            Data = data;
-            Width = DefaultWidth;
-            Color = color;
-        }
+        public Version Version => DependencyInfo.Version;
 
-        /// <summary>
-        /// Constructor
-        /// </summary>
-        /// <param name="data"></param>
-        /// <param name="width"></param>
-        /// <param name="brush"></param>
-        public ColumnData(string data, int width, Brush brush)
+        public string DetailsMessage
         {
-            Data = data;
-            Width = width;
-            Color = brush;
-        }
+            get
+            {
+                string message;
 
-        /// <summary>
-        /// Data in each cell
-        /// </summary>
-        public string Data
-        {
-            get; set;
-        }
+                switch (DependencyInfo.State)
+                {
+                    case PackageDependencyState.Loaded:
+                        message = string.Format("{0} {1} is installed locally.", 
+                            DependencyInfo.Name, DependencyInfo.Version.ToString());
+                        break;
+
+                    case PackageDependencyState.Missing:
+                        message = string.Format("This graph depends on {0} {1}. " +
+                            "Would you like to download and install this package?", 
+                            DependencyInfo.Name, DependencyInfo.Version.ToString());
+                        break;
+
+                    default:
+                        message = string.Format("Some nodes in this graph were originally created with {0} {1} " +
+                            "which you do not have installed locally. You may have a different version of {0} " +
+                            "installed, or you may have a different package installed which resolves these nodes.", 
+                            DependencyInfo.Name, DependencyInfo.Version.ToString());
+                        break;
+                }
 
-        /// <summary>
-        /// Width of each cell
-        /// </summary>
-        public int Width
-        {
-            get; set;
+                return message;
+            }
         }
 
-        /// <summary>
-        /// Foreground color of each cell
-        /// </summary>
-        public Brush Color
+        public ImageSource Icon
         {
-            get; set;
+            get
+            {
+                string iconPath;
+
+                switch (DependencyInfo.State)
+                {
+                    case PackageDependencyState.Loaded:
+                        iconPath = "NodeLibraryDependency_Loaded.png";
+                        break;
+
+                    case PackageDependencyState.Missing:
+                        iconPath = "NodeLibraryDependency_Missing.png";
```

Comment:
would it be possible to move these to a resx file? I guess we can do it as as followup - raw resources like this are difficult to manage and make patch installer issues much worse.

Reply:
for example look at:
https://github.com/DynamoDS/Dynamo/blob/master/src/Libraries/GeometryUIWpf/ProtoGeometryImages.resx

Reply:
@mjkkirschner I spent some time trying to figure this out last night, but couldn't quite get it to work. If it's alright with you, I'd like to deal with this in a follow up UI pass.

Reply:
@mjkkirschner Finally got it. Turns out we have an existing utility for converting bitmaps to ImageSources: [https://github.com/DynamoDS/Dynamo/blob/ec10f936824152e7dd7d6d019efdcda0d78a5264/src/DynamoCoreWpf/Utilities/ResourceUtilities.cs](https://github.com/DynamoDS/Dynamo/blob/ec10f936824152e7dd7d6d019efdcda0d78a5264/src/DynamoCoreWpf/Utilities/ResourceUtilities.cs)